src/flag_gems/runtime/register.py

```diff
@@ -60,20 +62,25 @@ def register_impl(self, key, fn, has_backward):
         )
         self.lib.impl(key, fn, device_key)
 
-    def for_each(self, config):
+    def for_each(self):
         try:
-            for key, func, has_backward in config:
+            for key, func, has_backward in self.config:
                 if key not in self.unused_ops:
                     self.register_impl(key, func, has_backward)
         except Exception as e:
             error.register_error(e)
 
-    def _set_info(self, config):
-        for _, fn, hasbackward in config:
+    def _set_info(self):
+        for _, fn, hasbackward in self.config:
             fn_name = fn.__name__
             self.backward_ops.append(
                 fn_name
-            ) if hasbackward else self.forward_ops.append(fn_name)
+            ) if hasbackward is commom_utils.Autograd.enable else self.forward_ops.append(
+                fn_name
```

Comment:
change this expression to a common conditional one.

Reply:
Done